packages/react-components/react-tag-picker/library/src/components/TagPickerControl/useTagPickerControl.tsx

```diff
@@ -122,4 +127,74 @@ export const useTagPickerControl_unstable = (
     disabled,
     invalid,
   };
+
+  const { targetDocument } = useFluent();
+  const win = targetDocument?.defaultView;
+
+  React.useEffect(() => {
+    if (!win || !triggerRef.current || !state.expandIcon) {return;}
+
+    // If aria-label or aria-labelledby changes, recalculate aria-label and aria-labelledby for the expandIcon
+    const getExpandLabel = (ariaLabel?: string | null, ariaLabelledBy?: string | null) => {
+      let expandAriaLabel = '';
+      let expandAriaLabelledBy = '';
+
+      if (state.expandIcon) {
+        const hasExpandLabel = state.expandIcon['aria-label'] || state.expandIcon['aria-labelledby'];
+        // If there is no explicit aria-label, calculate default accName attribute for expandIcon button,
```

Comment:
I might suggest moving this comment block above the entire useEffect, and update it slightly to clarify that it's basing the expandIcon's label off the input's label (in Combobox, the input was the primary slot so it didn't need to be specified)